seed/java-sdk/websocket-bearer-auth/src/main/java/com/seed/websocketBearerAuth/resources/realtime/websocket/RealtimeWebSocketClient.java

```diff
@@ -0,0 +1,379 @@
+            ClientOptions clientOptions, String id, Optional<String> model, Optional<Integer> temperature) {
+        this.clientOptions = clientOptions;
+        this.objectMapper = ObjectMappers.JSON_MAPPER;
+        this.okHttpClient = clientOptions.httpClient();
+        this.id = id;
+        this.model = model;
+        this.temperature = temperature;
+    }
+
+    /**
+     * Establishes the WebSocket connection asynchronously.
+     * @return a CompletableFuture that completes when the connection is established
+     */
+    public CompletableFuture<Void> connect() {
+        connectionFuture = new CompletableFuture<>();
+        String baseUrl = clientOptions.environment().getUrl();
+        StringBuilder pathBuilder = new StringBuilder();
+        pathBuilder.append("/realtime/");
+        pathBuilder.append(id);
+        pathBuilder.append("");
+        String fullPath = pathBuilder.toString();
+        if (baseUrl.endsWith("/") && fullPath.startsWith("/")) {
+            fullPath = fullPath.substring(1);
+        } else if (!baseUrl.endsWith("/") && !fullPath.startsWith("/")) {
+            fullPath = "/" + fullPath;
+        }
+        HttpUrl.Builder urlBuilder = HttpUrl.parse(baseUrl + fullPath).newBuilder();
+        if (model != null && model.isPresent()) {
+            urlBuilder.addQueryParameter("model", String.valueOf(model.get()));
+        }
+        if (temperature != null && temperature.isPresent()) {
+            urlBuilder.addQueryParameter("temperature", String.valueOf(temperature.get()));
+        }
+        Request.Builder requestBuilder = new Request.Builder().url(urlBuilder.build());
+        clientOptions.headers(null).forEach(requestBuilder::addHeader);
+        Request request = requestBuilder.build();
+        this.webSocket = okHttpClient.newWebSocket(request, new WebSocketListener() {
+            @Override
+            public void onOpen(WebSocket webSocket, Response response) {
+                if (onConnectedHandler != null) {
+                    onConnectedHandler.run();
+                }
+                connectionFuture.complete(null);
+            }
+
+            @Override
+            public void onMessage(WebSocket webSocket, String text) {
+                handleIncomingMessage(text);
+            }
+
+            @Override
+            public void onFailure(WebSocket webSocket, Throwable t, Response response) {
+                if (onErrorHandler != null) {
+                    onErrorHandler.accept(new RuntimeException(t));
+                }
+                connectionFuture.completeExceptionally(t);
+            }
+
+            @Override
+            public void onClosed(WebSocket webSocket, int code, String reason) {
+                if (onDisconnectedHandler != null) {
+                    onDisconnectedHandler.accept(new DisconnectReason(code, reason));
+                }
+            }
+        });
+        Executors.newSingleThreadScheduledExecutor()
```

Comment:
what shuts this down?

Instead of a CompletableFuture<Void>, should we return something that can be closed idiomatically via a try-with-resources ?